syntax = "proto3";

import "google/protobuf/timestamp.proto";

package chat;

message ChatMessage {
    string name = 1;
    string text = 2;   
    google.protobuf.Timestamp timestamp = 3;
}

message Empty {}

message ConnectRequest {
    string name = 1;
}

message ServerMeta {
    string server_name = 1;
    string motd = 2;
    int64 max_users = 3;
    int64 user_count = 4;
}

service ChatService {
    rpc Connect(ConnectRequest) returns (ServerMeta);
    rpc SendMessage(ChatMessage) returns (Empty);
    rpc ChatStream(Empty) returns (stream ChatMessage);
}
